### PR TITLE
fix(testenv): disable downloads (bitcoind and electrsd) for docs.rs b…

### PR DESCRIPTION
**Pull request by riverKanies**
…uilds of crate testenv

### Description

address this issue https://github.com/bitcoindevkit/bdk/issues/1627
where docs.rs build is failing for bdk_testenv crate

original PR: https://github.com/bitcoindevkit/bdk/pull/1679

### Notes to the reviewers

I was not able to reproduce the build issue locally, so this will have to be tested live unless someone else can reproduce the build error https://docs.rs/crate/bdk_testenv/0.10.0/builds/1377651

more details in this thread on discord https://discord.com/channels/753336465005608961/1265333904324427849/1304476756660719668

#### Bugfixes:

* [x] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Comment by ValuedMammal** [comment]
Will adding these fix the test error you're seeing?

<details open>
  <summary>diff</summary>
  
```diff
diff --git a/crates/bitcoind_rpc/Cargo.toml b/crates/bitcoind_rpc/Cargo.toml
index a3426007..6bf55262 100644
--- a/crates/bitcoind_rpc/Cargo.toml
+++ b/crates/bitcoind_rpc/Cargo.toml
@@ -20,9 +20,9 @@ bitcoin = { version = "0.32.0", default-features = false }
 bitcoincore-rpc = { version = "0.19.0" }
 bdk_core = { path = "../core", version = "0.3.0", default-features = false }
 
 [dev-dependencies]
-bdk_testenv = { path = "../testenv", default-features = false }
+bdk_testenv = { path = "../testenv" }
 bdk_chain = { path = "../chain" }
 
 [features]
 default = ["std"]
diff --git a/crates/electrum/Cargo.toml b/crates/electrum/Cargo.toml
index 03c3783b..b34a8d72 100644
--- a/crates/electrum/Cargo.toml
+++ b/crates/electrum/Cargo.toml
@@ -16,9 +16,9 @@ workspace = true
 bdk_core = { path = "../core", version = "0.3.0" }
 electrum-client = { version = "0.21", features = [ "proxy" ], default-features = false }
 
 [dev-dependencies]
-bdk_testenv = { path = "../testenv", default-features = false }
+bdk_testenv = { path = "../testenv" }
 bdk_chain = { path = "../chain" }
 
 [features]
 default = ["use-rustls"]
diff --git a/crates/esplora/Cargo.toml b/crates/esplora/Cargo.toml
index 2f42b4fd..3ebbd59c 100644
--- a/crates/esplora/Cargo.toml
+++ b/crates/esplora/Cargo.toml
@@ -22,9 +22,9 @@ futures = { version = "0.3.26", optional = true }
 miniscript = { version = "12.0.0", optional = true, default-features = false }
 
 [dev-dependencies]
 bdk_chain = { path = "../chain" }
-bdk_testenv = { path = "../testenv", default-features = false }
+bdk_testenv = { path = "../testenv" }
 tokio = { version = "1", features = ["rt", "rt-multi-thread", "macros"] }
 
 [features]
 default = ["std", "async-https", "blocking-https"]
diff --git a/crates/testenv/Cargo.toml b/crates/testenv/Cargo.toml
index 85551673..2b939b59 100644
--- a/crates/testenv/Cargo.toml
+++ b/crates/testenv/Cargo.toml
@@ -16,9 +16,12 @@ readme = "README.md"
 workspace = true
 
 [dependencies]
 bdk_chain = { path = "../chain", version = "0.20.0", default-features = false }
-electrsd = { version = "0.28.0", features = [ "legacy" ] }
+electrsd = { version = "0.28.0", features = [ "legacy" ], default-features = false }
+
+[dev-dependencies]
+bdk_testenv = { path = "." }
 
 [features]
 default = ["std", "download"]
 download = ["electrsd/bitcoind_25_0", "electrsd/esplora_a33e97e1"]
```

</details>

**Comment by riverKanies** [comment]
@ValuedMammal that did fix the test! At first I thought this might have unwanted side effects, but after some time I do think this is the solution we want!

P.S. originally I was only able to confirm tests on explora as the tests for other clients were failing for me on master with: `Error: JSON-RPC error: transport error: Couldn't connect to host: Resource temporarily unavailable (os error 35)`
I found a solution and created issue https://github.com/bitcoindevkit/bdk_wallet/issues/43

all tests are now passing for me locally with your proposed changes VM 🥳

File: crates/testenv/Cargo.toml [hunk 1]
```diff
@@ -17,9 +17,16 @@ workspace = true
...
-electrsd = { version = "0.28.0", features = [ "bitcoind_25_0", "esplora_a33e97e1", "legacy" ] }
+electrsd = { version = "0.28.0", features = [ "legacy" ], default-features = false }
+
+[dev-dependencies]
+bdk_testenv = { path = "." }
 
 [features]
-default = ["std"]
+default = ["std", "download"]
+download = ["electrsd/bitcoind_25_0", "electrsd/esplora_a33e97e1"]
```
Comment (ValuedMammal) [inline crates/testenv/Cargo.toml:27]: We probably don't need a new named feature per se, but instead just include the desired features in "default". What do you think?

```suggestion
default = ["std", "electrsd/bitcoind_25_0", "electrsd/esplora_a33e97e1"]
```

Comment (riverKanies) [reply crates/testenv/Cargo.toml:27]: personally I think it's a bit more clear what's going on with the explicit 'download' feature. I would even add a comment maybe since this issue is a bit hard to parse and the solution isn't super direct (it's not super obvious that the purpose of the download feature is to allow disabling downloads in doc builds indirectly by disabling default features)

**Comment by notmandatory** [comment]
Can you just squash the 2nd commit into the first ? then looks looks good to go. Thanks!

**Comment by riverKanies** [comment]
@notmandatory squashed and rebased to master

**Comment by ValuedMammal** [comment]
Are you signing git commits also?

https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key

**Comment by riverKanies** [comment]
> Are you signing git commits also?
> 
> https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key

hmm never heard of this. looks like i managed to set it up, now my commits should be tagged as verified 👍

File: crates/testenv/Cargo.toml [hunk 2]
```diff
@@ -17,9 +17,16 @@ workspace = true
 
 [dependencies]
 bdk_chain = { path = "../chain", version = "0.20.0", default-features = false }
-electrsd = { version = "0.28.0", features = [ "bitcoind_25_0", "esplora_a33e97e1", "legacy" ] }
+electrsd = { version = "0.28.0", features = [ "legacy" ], default-features = false }
+
+[dev-dependencies]
+bdk_testenv = { path = "." }
```
Comment (oleonardolima) [inline crates/testenv/Cargo.toml:23]: Do we need this change? 🤔

**Review by oleonardolima — COMMENTED** [review]
I've been trying to reproduce the problem locally by running [doc.rs](https://github.com/rust-lang/docs.rs?tab=readme-ov-file#getting-started) locally, but got no success 😢.

It looks pretty straight-forward, and at least running `cargo doc` locally worked just fine.

File: crates/testenv/Cargo.toml (hunk 2 above)
Comment (riverKanies) [reply crates/testenv/Cargo.toml:23]: hmm, I actually don't know what it does, and tests run fine without it. I just left it in since it was in @ValuedMammal 's suggested changes and didn't hurt anything. appearently it's a 'common pattern' but ya idk what purpose it serves. dev-dependencies are only for tests right?

Comment (ValuedMammal) [reply crates/testenv/Cargo.toml:23]: The rationale was that it effectively enables default features for the single test in testenv `test_reorg_is_detected_in_electrsd`, but it's possible we won't notice a difference in CI where we test the whole workspace

Comment (oleonardolima) [reply crates/testenv/Cargo.toml:23]: Oh, got it. I was testing the whole workspace locally and noticed it wasn't required, that's why I asked.

Comment (riverKanies) [reply crates/testenv/Cargo.toml:23]: ah, interesting!

**Review by ValuedMammal — APPROVED** [review]
ACK 9b48fd43fe203ee6dc7451264768093d9cc61297 but I guess we won't know for sure until we publish the crate

**Review by notmandatory — APPROVED** [review]
ACK 9b48fd43fe203ee6dc7451264768093d9cc61297

**Comment by notmandatory** [comment]
Docs for this crate are now being published! 🎉 

https://docs.rs/bdk_testenv/0.11.0/bdk_testenv/

**Comment by riverKanies** [comment]
great to confirm this! I can cross it off my list
